src/components/charts/BarChart.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div class="bar-chart-container" v-if="loaded">
+    <Bar
+      id="my-chart-id"
+      :options="chartOptions"
+      :data="chartData"
+    />
+  </div>
+  <div v-else>
+    Loading...
+  </div>
+</template>
+
+<script>
+import { Bar } from 'vue-chartjs';
+import { Chart as ChartJS, Title, Tooltip, Legend, BarElement, CategoryScale, LinearScale } from 'chart.js';
+
+ChartJS.register(Title, Tooltip, Legend, BarElement, CategoryScale, LinearScale);
+
+export default {
+  name: 'BarChart',
+  components: { Bar },
+  data() {
+    return {
+      loaded: false,
+      chartOptions: {
+        animation: false,
+        transitions: {
+          active: {
+            animation: {
+              duration: 0
+            }
+          }
+        },
+        responsive: true,
+        maintainAspectRatio: false,
+        scales: {
+          x: {
+            stacked: true
+          },
+          y: {
+            stacked: true
+          }
+        },
+        plugins: {
+          title: {
+            display: true,
+            text: "Croom" // Default title
+          }
+        }
+      },
+      chartData: {
+        labels: ['Free', 'Occupied'],
+        datasets: [
+          {
+            label: 'Bed Status',
+            data: [],  
+            backgroundColor: ['rgba(75, 192, 192, 0.2)', 'rgba(255, 99, 132, 0.2)'],
+            borderColor: ['rgba(75, 192, 192, 1)', 'rgba(255, 99, 132, 1)'],
+            borderWidth: 1
+          }
+        ]
+      },
+    };
+  },
+ 
+  methods: {
+    async countDisabledBeds() {
+      try {
+        const response = await fetch('/api/beds/statuses?hospital_id=1'); // Adjust hospital_id
+        const data = await response.json();
+  
+        if (!Array.isArray(data)) {
+          throw new Error("Invalid data format");
+        }
+        
+        const disabledCount = data.filter(item => item.disabled).length;
+        const enabledCount = data.length - disabledCount; 
+  
+        this.chartData.datasets[0].data = [enabledCount, disabledCount];
+  
+        this.loaded = true;
+
+        if (this.chartData.datasets[0].data.length > 0) {
+          console.log("Chart data is populated:", this.chartData.datasets[0].data);
+        } else {
+          console.log("Chart data is empty or not yet populated.");
+        }
+      } catch (error) {
+        console.error("Error fetching data:", error);
+        this.loaded = true;
+      }
+    },
+    
+    async fetchHospitalName() {
```

Comment:
This will no longer be nessescary as the prop hospital will contain this information